### PR TITLE
add reduce_labels_to_label_edges + test

### PR DESCRIPTION
**Pull request by haesleinhuepf**
Hey @StRigaud ,

this adds another function `reduce_labels_to_label_edges`. It seems the test is not executed though:

```
C:\structure\code\CLIc>C:\structure\code\CLIc\out\build\win-msvc17\tests\tier2\Debug\test_reduce_labels_to_label_edges.exe
Running main() from C:\structure\code\CLIc\out\build\win-msvc17\_deps\googletest-src\googletest\src\gtest_main.cc
[==========] Running 0 tests from 0 test suites.
[==========] 0 tests from 0 test suites ran. (0 ms total)
[  PASSED  ] 0 tests.
```

Maybe you see immediately, why.

Otherwise, this was a nice exercise which 
closes https://github.com/clEsperanto/pyclesperanto/issues/139 

Cheers,
Robert

### CODE REVIEW

**Comment by StRigaud** [comment]
>  It seems the test is not executed though:

I gues it is still related to the gmock library that we have fixed in #246 but because you branched from main you dont have the modification yet
target_link_libraries(${test_name} PRIVATE CLIc::CLIc gtest_main gmock_main)

File: tests/tier2/test_reduce_labels_to_label_edges.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#include "cle.hpp"
+
+#include <array>
+#include <gtest/gtest.h>
+
+class TestReduceLabelsToLabelEdges : public ::testing::TestWithParam<std::string>
+{
+protected:
+  std::array<uint32_t, 8 * 8 * 1> output;
+s
```
Comment (StRigaud) [inline tests/tier2/test_reduce_labels_to_label_edges.cpp:10]: typo

Comment (StRigaud) [reply tests/tier2/test_reduce_labels_to_label_edges.cpp:10]: The test still fail, but at least the compilation works once the typo is gone

File: tests/tier2/test_reduce_labels_to_label_edges.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+TEST_P(TestReduceLabelsToLabelEdges, execute)
+{
+  std::string param = GetParam();
+  cle::BackendManager::getInstance().setBackend(param);
+  auto device = cle::BackendManager::getInstance().getBackend().getDevice("", "all");
+  device->setWaitToFinish(true);
+
+  auto gpu_input = cle::Array::create(5, 4, 2, 3, cle::dType::FLOAT, cle::mType::BUFFER, device);
+  gpu_input->write(input.data());
```
Comment (StRigaud) [inline tests/tier2/test_reduce_labels_to_label_edges.cpp:27]: The `gpu_input` `Array::create()` does not correspond to the array CPU you are trying to process

`output` is of type `uint8` of shape `8*8*1`
`valid` and `input` are of type `float`
and you create `gpu_input` as an array of `(5,4,2)` of type `float`

Comment (StRigaud) [reply tests/tier2/test_reduce_labels_to_label_edges.cpp:27]: I let you fix the test :)

**Comment by haesleinhuepf** [comment]
🎉

**Review by StRigaud — APPROVED** [review]

**Pull request merged**